### ISSUE TITLE
vSphere Multiple Networks - Rancher uses Random IPs for SSH

### ISSUE THREAD

**Issue by TimUx** [rancher/rancher#32205]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
Add Cluster with multiple Nodes, which have miltiple Network Interfaces in Node-Template

**Result:**
Some Nodes are in Status "Waiting for SSH"

**Other details that may be helpful:**
The SSH Prozesses at the Rancher VM shows, that the SSH IP is wrong.
At This YouTobe Video ( passage ) https://youtu.be/ozLPpyrqwf8?t=1234 is explained, that Rancher will use the first open-vm-tools reported IP.
But in some Nodes, Rancher uses randomly IPs, not the first but one of the other IPs.
Rancher can't connect ( routing, firewall, ... ) to the other three IPs, just to the first one.

We've konfigured the the open-vm-tools, to report the IP of specific interface first .

/etc/vmware-tools/tools.conf

```
    [guestinfo]
    primary-nics=ens224
    low-priority-nics== ens256,ens161,ens193
```

This works very well und the right IP is everytime at every VM at the first position.
Rancher still takes the IPs randomly.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): rancher/rancher v2.5.7
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VM
- Kubernetes version (use `kubectl version`): 

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"clean", BuildDate:"2021-03-18T01:10:43Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:03:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.13.8
 Git commit:        afacb8b7f0
 Built:             Fri Dec 18 12:15:19 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       afacb8b7f0
  Built:            Fri Dec  4 23:02:49 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.3.3-0ubuntu2.3
  GitCommit:
 runc:
  Version:          spec: 1.0.1-dev
  GitCommit:
 docker-init:
  Version:          0.18.0
  GitCommit:

```

**Comment by nishiokay** [comment]
Hi @TimUx ,
We faced the same issue.
I checked the source code of the ranch-machine and created a patch file.
https://gist.github.com/nishiokay/119c7b9ba680c329baac068d4e38834f
https://pkg.go.dev/github.com/vmware/govmomi/object#VirtualMachine.WaitForNetIP

With this patch, the rancher-machine will attempt an ssh connection using the IP address of the first NIC.
This patch is working fine in our environment.

I will send the PR later.
